host/hackrf-tools/src/hackrf_debug.c

```diff
@@ -417,35 +417,49 @@ int rffc5072_write_register(
 	return result;
 }
 
-int fpga_spi_read_register(hackrf_device* device, const uint16_t register_number)
+int fpga_read_register(hackrf_device* device, const uint16_t register_number)
 {
 	uint8_t register_value;
-	int result =
-		hackrf_fpga_spi_read(device, (uint8_t) register_number, &register_value);
+	int result = hackrf_fpga_read(device, (uint8_t) register_number, &register_value);
 
 	if (result == HACKRF_SUCCESS) {
 		printf("[%2d] -> 0x%02x\n", register_number, register_value);
 	} else {
-		printf("hackrf_fpga_spi_read() failed: %s (%d)\n",
+		printf("hackrf_fpga_read() failed: %s (%d)\n",
 		       hackrf_error_name(result),
 		       result);
 	}
 
 	return result;
 }
 
-int fpga_spi_write_register(
+int fpga_read_registers(hackrf_device* device)
+{
+	uint16_t register_number;
+	int result = HACKRF_SUCCESS;
+
+	for (register_number = 1; register_number <= 5; register_number++) {
```

Comment:
I left out the bulk read function for FPGA registers intentionally: we don't know the number of registers any gateware might have. I'm still not sure about this.

Reply:
We could find a way to do it, yeah